Circuit/Components/VacuumTubes/Triode.cs

```diff
@@ -138,14 +140,23 @@ public override void Analyze(Analysis Mna)
             {
                 case TriodeModel.ChildLangmuir:
                     Expression Ed = Mu * Vgk + Vpk;
-                    ip = Call.If(Ed > 0, K * (Ed ^ 1.5), 0);
+                    ip = Call.If(Ed > 0, K * Binary.Power(Ed, 1.5), 0);
                     ig = 0;
                     ik = -(ip + ig);
                     break;
                 case TriodeModel.Koren:
                     Expression E1 = Ln1Exp(Kp * (1.0 / Mu + Vgk * Binary.Power(Kvb + Vpk * Vpk, -0.5))) * Vpk / Kp;
-                    ip = Call.If(E1 > 0, (E1 ^ Ex) / Kg, 0);
-                    ig = Call.Max(Vgk - Vg, 0) / Rgk;
+                    ip = Mna.AddUnknownEqualTo(Call.If(E1 > 0, 2d * (E1 ^ Ex) / Kg, 0));
+
+                    var vg = (double)Vg;
+                    var knee = (double)Kn;
+                    var rg1 = (double)Rgk;
+
+                    var a = 1 / (4 * knee * rg1);
+                    var b = (knee - vg) / (2 * knee * rg1);
+                    var c = (-a * Math.Pow(vg - knee, 2)) - (b * (vg - knee));
+
+                    ig = Mna.AddUnknownEqualTo(Call.If(Vgk < vg - knee, 0, Call.If(Vgk > vg + knee, (Vgk - vg) / rg1, a * Vgk * Vgk + b * Vgk + c)));
```

Comment:
(Not directly related to this PR) Should the `SimulateCapacitances` branch below be outside this `switch` so it applies to all the models?

Reply:
Probably they should 😉

Reply:
Done